lib/spack/docs/build_settings.rst

```diff
@@ -179,15 +179,72 @@ Specific limitations include:
 
 * A package must define ``executables`` and ``determine_spec_details``
   for Spack to locate instances of that package.
-* This is currently intended to find build dependencies rather than
-  library packages.
+* This implementation only collects and examines executable files, so it
+  is only useful for build/run dependencies.
+* The logic does not search through modules module files, it can only
+  detect packages with executables defined in ``PATH``
 * Spack does not overwrite existing entries in the package configuration:
   If there is an external defined for a spec at any configuration scope,
   then Spack will not add a new external entry (``spack config blame packages``
   can help locate all external entries).
-* Currently this logic is focused on examining ``PATH`` and does not
-  search through modules (although it should find the package if a
-  module is loaded for it).
+
+As mentioned above, to make a package discoverable with
+``spack external find`` a user must define one or more executables
+associated with the package and must implement a method to generate
+a Spec when given an executable.
+
+The executables are specified as a package level ``executables``
+attribute which is a list of strings (full example with context given
+below); each string is treated as a regular expression (e.g. 'gcc'
+would match 'gcc', 'gcc-8.3', 'my-weird-gcc', etc.).
+
+The method, ``determine_spec_details``, has the following signature:
```

Comment:
The docs below look good, though the style is different from the rest of the spack docs.  These should be in second person and casual, e.g., "To enable your package to find externals, you can add a method call `determine_spec_details`, like this:"

etc.